### PR TITLE
Add Sphinx documentation for the super-build

### PR DESCRIPTION
**Pull request by ghost**
Mostly moved and reorganised from its original home in ome-files/bioformats, updated with building instructions for the split components, and updated maintenance instructions for the superbuild packages.

See also: https://github.com/ome/ome-files-cpp/pull/17

---

For both PRs, you can build the docs without doing a C++ build by:

```
cd docs/sphinx
make [man] html pdf linkcheck
```

and the results will be in the `build` directory.  They will also be present in `share/doc/<pkg>` in the superbuild artefacts.

### CODE REVIEW

**Comment by sbesson** [comment]
Note this PR also contains the proposal of switching back to `.rst` as the default extension which might need to be applied across our repos /cc @hflynn

**Comment by ghost** [comment]
I should note that while I'd like to leave the superbuild-specific stuff here, there is potential for moving the `packages/*` files out to a separate repository which can be referenced from multiple documentation sets.

**Comment by sbesson** [comment]
@rleigh-dundee: what is the amount of Sphinx documentation that would remain at the level of the individual components e.g. `ome-files-cpp`?

**Comment by ghost** [comment]
@sbesson I would think it would be limited to component-specific documentation: overview of what the component is for, minimal build instructions (cmake options), schema information, examples/tutorials.  We can repeat this for the other components as well, where needed.

**Comment by ghost** [comment]
@sbesson You can see the changes in the second PR.  The ome-files docs have basically lost the building instructions from the overview page and the maintenance instructions from the separate maintenance page.  I've kept a minimal set of build options documented in ome-files, but the full set is now here.

File: docs/sphinx/conf.py.in [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+jenkins_view_root = jenkins_root + '/view'
+
+# Variables used to define other extlinks
+cvs_root = 'http://cvs.openmicroscopy.org.uk'
+trac_root = 'https://trac.openmicroscopy.org/ome'
+oo_root = 'http://www.openmicroscopy.org'
+oo_site_root = oo_root + '/site'
+lists_root = 'http://lists.openmicroscopy.org.uk'
+downloads_root = 'http://downloads.openmicroscopy.org'
+omerodoc_uri = oo_site_root + '/support/omero5.1'
```
Comment (hflynn) [inline docs/sphinx/conf.py.in]: this should probably just be /omero/ so it auto-redirects to the latest version. If that's going to cause issues, it should at least be 5.2

Comment (ghost) [inline docs/sphinx/conf.py.in]: These just got copied in from bioformats' sphinx configuration.  We aren't using these, so we could just delete them.

File: docs/sphinx/themes/globalbftoc.html [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<h3><a href="{{ pathto(master_doc) }}">{{ _('Bio-Formats') }}</a></h3>
+{{ toctree()}}
+<a href="http://downloads.openmicroscopy.org/latest/bio-formats5.2/">{{ _('Bio-Formats Downloads') }}</a></br>
```
Comment (hflynn) [inline docs/sphinx/themes/globalbftoc.html]: Not sure we need Bf download links here

File: docs/sphinx/conf.py.in (hunk 1 above)
Comment (hflynn) [inline docs/sphinx/conf.py.in]: 👍 to getting rid of unused extlinks

**Comment by hflynn** [comment]
Haven't had time to review the structure but a browse through the content looks ok.

**Comment by sbesson** [comment]
Note that although https://ci.openmicroscopy.org/view/Failing/job/OME-FILES-CPP-DEV-merge-win-superbuild/ is failing, a [previous build](https://ci.openmicroscopy.org/job/OME-FILES-CPP-DEV-merge-win-superbuild/292/) passed and generated the Sphinx documentation on Windows. Still investigating the cause of these failures.

**Comment by ghost** [comment]
The failure is due to https://github.com/ome/ome-qtwidgets/pull/8 and I'll have a fix for it shortly.  That's all part of unifying and simplifying the build to match the documentation.  It needs a tweak to the jenkins-build.bat script to make it find the Qt libs.

**Comment by sbesson** [comment]
Temporarily excluding to get the cmake PRs passing.

**Comment by sbesson** [comment]
Rechecked the documentation sets build on Windows using the binaries now the builds are green again. There are a couple of broken links to the doxygen documentation but https://trello.com/c/qMsuMTM9/19-adjust-download-directory-contents is the next step to simplify the maintenance of these links. Merging.

**Pull request merged**